src/Features/CSharp/Portable/CodeRefactorings/AddNamedArguments/CSharpAddNamedArgumentsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,221 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CodeRefactorings.AddNamedArguments;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.AddNamedArguments
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(CSharpAddNamedArgumentsCodeRefactoringProvider)), Shared]
+    internal class CSharpAddNamedArgumentsCodeRefactoringProvider : AbstractAddNamedArgumentsCodeRefactoringProvider<ArgumentSyntax>
+    {
+        protected override ArgumentSyntax[] GetNamedArguments(SeparatedSyntaxList<ArgumentSyntax> arguments, ImmutableArray<IParameterSymbol> parameters, ref bool hasLiteral)
+        {
+            // Early return when already fixed
+            if (arguments.Count > 0 && arguments[0].NameColon != null)
+            {
+                return null;
+            }
+
+            var result = ArrayBuilder<ArgumentSyntax>.GetInstance();
+            for (int index = 0; index < arguments.Count; ++index)
+            {
+                var argument = arguments[index];
+                if (argument.NameColon != null)
+                {
+                    result.Add(argument);
+                }
+                else
+                {
+                    var parameter = parameters[index];
+                    var argumentName = NameColon(parameter.Name);
+                    if (parameter.IsParams && arguments.Count - index > 1)
+                    {
+                        var arrayBuilder = ArrayBuilder<ExpressionSyntax>.GetInstance(arguments.Count - index);
+                        for (; index < arguments.Count; ++index)
+                        {
+                            arrayBuilder.Add(arguments[index].Expression);
+                        }
+
+                        var expressions = arrayBuilder.ToArrayAndFree();
+                        var initializer = InitializerExpression(SyntaxKind.ArrayInitializerExpression, SeparatedList(expressions));
+                        var arrayType = ArrayType(ParseTypeName(parameter.Type.ToDisplayString()));
```

Comment:
@CyrusNajmabadi Changing this line to `ArrayType(parameter.Type.GenerateTypeSyntax());` is enough? Because currently I don't have any of those objects in this method to get SyntaxGenerator